core/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java

```diff
@@ -301,6 +296,22 @@ private Path unzip(Path zip, Path pluginsDir) throws IOException, UserError {
         return target;
     }
 
+    private Path stagingDirectory(Path pluginsDir) throws IOException {
+        try {
+            Set<PosixFilePermission> perms = new HashSet<>();
+            perms.add(PosixFilePermission.OWNER_EXECUTE);
+            perms.add(PosixFilePermission.OWNER_READ);
+            perms.add(PosixFilePermission.OWNER_WRITE);
+            perms.add(PosixFilePermission.GROUP_READ);
+            perms.add(PosixFilePermission.GROUP_EXECUTE);
+            perms.add(PosixFilePermission.OTHERS_READ);
+            perms.add(PosixFilePermission.OTHERS_EXECUTE);
+            return Files.createTempDirectory(pluginsDir, ".installing-", PosixFilePermissions.asFileAttribute(perms));
+        } catch (IllegalArgumentException | UnsupportedOperationException e) {
+            return Files.createTempDirectory(pluginsDir, ".installing-");
```

Comment:
IAE only happens if the prefix (`.installing-`) cannot be used. So if it fails, we shouldnt try the same prefix again, instead pass null?